libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/routeline/RouteLineComponent.kt

```diff
@@ -69,14 +72,21 @@ internal class RouteLineComponent(
         }
 
         coroutineScope.launch {
-            mapboxNavigation.flowRoutesUpdated().collect { result ->
-                val routeLines = result.navigationRoutes.map {
-                    NavigationRouteLine(it, null)
-                }
-                routeLineApi.setNavigationRouteLines(routeLines).let { routeDrawData ->
-                    ifNonNull(mapView.getMapboxMap().getStyle()) { style ->
-                        routeLineView.renderRouteDrawData(style, routeDrawData)
+            val routesFlow = mapboxNavigation.flowRoutesUpdated()
+                .map { it.navigationRoutes }
+                .stateIn(
+                    this,
+                    SharingStarted.WhileSubscribed(),
```

Comment:
Why is this needed? 🤔 . Does it change anything?

Reply:
Discussed in slack, converts it into a stateFlow 👍 

they way i did this in another place is the `.onStart { emit(..) }` operator